compiler/rustc_const_eval/src/check_consts/qualifs.rs

```diff
@@ -147,6 +147,11 @@ impl Qualif for NeedsDrop {
     }
 
     fn in_any_value_of_ty<'tcx>(cx: &ConstCx<'_, 'tcx>, ty: Ty<'tcx>) -> bool {
+        // Avoid selecting for simple cases, such as builtin types.
+        if ty::util::is_trivially_const_drop(ty) {
+            return false;
+        }
+
         ty.needs_drop(cx.tcx, cx.typing_env)
```

Comment:
Doesn't `ty.needs_drop` call `is_trivially_const_drop`? Maybe it should?

Reply:
It's clearly important for perf since not doing this caused the regression?

Reply:
I'd rather not have the same `is_trivially_const_drop() || needs_drop()` duplicated everywhere. You are already duplicating it twice in this PR.

Reply:
Looking at the implementation of `is_trivially_const_drop` is should really be renamed to `is_trivially_no_drop`.

Reply:
If you look back in the history, the thing that made perf neutral was actually checking ty.needs_drop lol, so I don't think it's necessary.

Reply:
So can you remove the calls to `is_trivially_const_drop` here then?